File: app/src/main/java/com/tj/newsreader/models/Article.java
package com.tj.newsreader.models;

import com.squareup.moshi.Json;

import java.util.List;


public class Article {

//name,title,description,Image


    @Json(name = "status")
    private String status;
    @Json(name = "totalResults ")
    private int totalResults;
    @Json(name = "articles")
    private List<NewsArticle> articles = null;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;

    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }

    public List<NewsArticle> getArticles() {
        return articles;
    }

    public void setArticles() {
        setArticles();
    }

    public void setArticles(List<NewsArticle> articles) {
        this.articles = articles;
    }
}

File: app/src/main/java/com/tj/newsreader/models/Article.kt
package com.tj.newsreader.models

import com.squareup.moshi.Json


class Article {

    //name,title,description,Image


    @Json(name = "status")
    var status: String? = null
    @Json(name = "totalResults ")
    private var totalResults: Int = 0
    @Json(name = "articles")
    var articles: List<NewsArticle>? = null

    fun getTotalResults(): Int? {
        return totalResults
    }

    fun setTotalResults(totalResults: Int?) {
        this.totalResults = totalResults!!
    }

    fun setArticles() {
        setArticles()
    }
}











